tests/unit/test_image.py

```diff
@@ -246,6 +246,25 @@ def test_train(_download_folder, _cleanup, popen, _stream_output, LocalSession,
                 assert config['services'][h]['command'] == 'train'
 
 
+@patch('sagemaker.local.local_session.LocalSession')
+@patch('sagemaker.local.image._stream_output', side_effect=RuntimeError('this is expected'))
+@patch('subprocess.Popen')
+@patch('sagemaker.local.image._SageMakerContainer._cleanup')
+@patch('sagemaker.local.image._SageMakerContainer._download_folder')
+def test_train_error(_download_folder, _cleanup, popen, _stream_output, LocalSession, tmpdir, sagemaker_session):
+    directories = [str(tmpdir.mkdir('container-root')), str(tmpdir.mkdir('data'))]
+
+    with patch('sagemaker.local.image._SageMakerContainer._create_tmp_folder', side_effect=directories):
+        instance_count = 2
+        image = 'my-image'
+        sagemaker_container = _SageMakerContainer('local', instance_count, image, sagemaker_session=sagemaker_session)
+
+        with pytest.raises(RuntimeError) as e:
```

Comment:
I dont think this should be one single test. I would split it to be a serving_container_handles_failure  or something.

Reply:
I'm not entirely sure what you mean here. `_stream_output` is patched to return a `RuntimeError`, so all this is testing is that the eventual `RuntimeError` returned by `train()` contains the error message from the first `RuntimeError`